src/Platform/Vulkan/VulkanDevice.hh

```diff
@@ -26,7 +26,7 @@ namespace esp
 
     ~VulkanDevice();
 
-    static inline VkDevice get_logical_device() { return s_instance->m_device; }
+    static inline VkDevice& get_logical_device() { return s_instance->m_device; }
```

Comment:
@DawidSzpejna if we want to return a reference here, it has to be const to avoid changing it's data